docs/api.md

```diff
@@ -6,6 +6,206 @@ challenges presented by JavaScript and web browsers.
+
+- `overwrite` (optional, default `false`): if `true`, will overwrite whatever is
+  at `fileName` with `data`. If `false`, will throw an error and halt the
+  subsequent enqueued evaluation (similar to a syntax error).
+
+
+#### `iodide.file.save` patterns
+
+```javascript
+
+// upload a csv to the server. Assume the object csvData is an array of objects
+// representing a columnar dataset. This overwrites whatever is in `cached-data.csv`
+// because overwrite is set to true.
+
+const csvData = [{x1: 10, x2: 'test1'}, {x1: 20, x2: 'test2'}, ...]
+
+iodide.file.save(csvData, 'cached-data.csv' { overwrite: true })
+
+// iodide.file.save works with ArrayBuffers as well.
+iodide.file.save(new ArrayBuffer(...), 'model-output.bin')
+```
+
+
+
+### `iodide.file.load(fileName, fileType[, variableName])`
+
+Returns a
+[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+that, when resolved, loads into the notebook the save file `fileName`. For each
+notebook can find a list of uploaded files in the [upload modal](#link-required)
+or on your notebook's [Revisions Page](#needs-link). You can also access any
+uploaded file through the fetch chunk, [following this
+pattern](https://iodide-project.github.io/docs/workflows/#uploading-data-to-an-iodide-notebook).
+For most use cases using [fetch
+chunks](https://iodide-project.github.io/docs/jsmd/#fetch-chunks-fetch) is
+preferable and more straightforward. In more dynamic programmatic cases,
+however, `iodide.file.load` can provide more nuanced workflows.
+
+`fileName` is the name of the file uploaded to the Iodide server. 
+
+`fileType` is the file type to handle. These are the same as the file types in
+[fetch chunks](https://iodide-project.github.io/docs/jsmd/#fetch-chunks-fetch):
+`js` (load this file as javascript, eg a library), `css` (load this file as a
+stylesheet, applying the styles to the report), `json` (load this file as json
+and parse into a javascript object), `text` (load this file as text), `blob`
+(load this file as binary).
+
+`variableName` (required for `json`, `text`, and `blob` file types, otherwise
+not applicable): the variable name in which to load the data, available in the
+browser `window` namespace.
+
+
+#### `iodide.file.load` patterns
+
+```javascript
+
+// load a csv
+iodide.file.load('cached-data.csv', 'text', 'cachedData').then(() => {
+  cachedData = d3.csvParsed(cachedData)
+})
+
+// load a video
+iodide.file.load('gritty.mp4', 'blob', 'gritty').then(() => {
+  // load this gritty video into a <video /> tag in the report.
+  var urlCreator = window.URL || window.webkitURL;
+  var imageUrl = urlCreator.createObjectURL(gritty);
+  document.querySelector("#gritty").src = imageUrl;
+})
+
+// load a style sheet and apply the style
+iodide.file.load('corporate-report-style.css', 'css')
+
+// load some javascript helpers
+iodide.file.load('analysis-helpers.js', 'js')
+```
+
+
+
+### `iodide.file.delete(fileName)`
```

Comment:
does anything happen if you attempt to delete and there isn't a file present?

Reply:
yes, it throws an error, which can be catched in the return value Promise with `.catch`. I'll include an example that shows this.